src/common/core/core.js

```diff
@@ -3488,7 +3524,8 @@ define([
         this.addLibrary = function (node, name, libraryRootHash, libraryInfo, callback) {
             ensureType(callback, 'callback', 'function');
             var error = ensureNode(node, 'node', true);
-            error = error || ensureType(name, 'name', 'string');
+            error = error || ensureType(name, 'name', 'string', true);
+            error = error || ensureMongoCompatibleKey(name, 'name', false, true);
```

Comment:
Is it really needed for libraries?

Reply:
For libraries, most of the things (like '.') were already checked, but it made things more consistent. Also, it is necessary as the name of the library will be the name attribute of the root node of the library subtree

Reply:
:+1: